pallets/settlement-zksync/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+    }
+
+    pub mod config_preludes {
+        #[cfg(test)]
+        pub use testing::*;
+
+        #[cfg(test)]
+        mod testing {
+            use frame_support::pallet_prelude::*;
+            use frame_support::{derive_impl, register_default_impl};
+
+            pub struct TestDefaultConfig;
+
+            #[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig, no_aggregated_types)]
+            impl frame_system::DefaultConfig for TestDefaultConfig {
+                #[inject_runtime_type]
+                type RuntimeEvent = ();
+            }
+
+            #[register_default_impl(TestDefaultConfig)]
+            impl crate::pallet::DefaultConfig for TestDefaultConfig {
+                type WeightInfo = ();
+            }
+        }
+    }
+
+    pub fn verify_proof<T: Config>(full_proof: Proof) -> Result<(), Error<T>> {
+        let pubs_bytes = &full_proof[PROOF_SIZE..];
+        let proof_bytes = &full_proof[..PROOF_SIZE];
+        let pubs = <[u8; PUBS_SIZE]>::try_from(pubs_bytes)
+            .map_err(|e| log::error!("Cannot get pubs data: {:?}", e))
+            .map_err(|_| Error::<T>::InvalidInput)?;
+        let raw_proof = <[u8; PROOF_SIZE]>::try_from(proof_bytes)
+            .map_err(|e| log::error!("Cannot get raw proof data: {:?}", e))
+            .map_err(|_| Error::<T>::InvalidInput)?;
+
+        native::zksync_verify::verify(&raw_proof, pubs)
+            .map(|_x| T::OnProofVerified::on_proof_verified(compute_hash(pubs_bytes)))
+            .map_err(Into::into)
+    }
+
+    impl<T> From<native::VerifyError> for Error<T> {
+        fn from(e: native::VerifyError) -> Self {
+            match e {
+                native::VerifyError::InvalidInput => Error::<T>::InvalidInput,
+                native::VerifyError::InvalidProofData => Error::<T>::InvalidProofData,
+                native::VerifyError::VerifyError => Error::<T>::VerifyError,
+            }
+        }
+    }
+
+    // Dispatchable functions allows users to interact with the pallet and invoke state changes.
+    // These functions materialize as "extrinsics", which are often compared to transactions.
+    // Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+    #[pallet::call(weight(<T as Config>::WeightInfo))]
+    impl<T: Config> Pallet<T> {
+        /// Submit a new ZkSync-Era proof.
+        #[pallet::call_index(0)]
+        pub fn submit_proof(
+            _origin: OriginFor<T>,
+            raw_proof: Box<Proof>,
+        ) -> DispatchResultWithPostInfo {
+            log::trace!("Submitting proof");
+            verify_proof::<T>(*raw_proof)
+                .map(Into::into)
+                .map_err(Into::into)
+        }
+    }
+
+    const PREFIX: &[u8; 7] = b"zksync-";
+    fn compute_hash(pubs: &[u8]) -> H256 {
+        let mut data_to_hash = PREFIX.to_vec();
+        data_to_hash.extend_from_slice(&pubs);
+        H256(keccak_256(data_to_hash.as_slice()))
+    }
```

Comment:
Not directly related to your changes, but I see test `fflonk_hash_as_expected` can be removed, being already covered by `valid_proof_passes_verification_and_is_notified` (`pallets/settlement-fflonk/src/tests.rs`).

Reply:
Maybe is better to cover it in a new PR